File: opcua-webservice-spring/examples/uawebservicedemo/src/main/java/org/opcfoundation/uawebservicedemo/models/company/skillclassification/SkillClassificationSubmodelType.java
package org.opcfoundation.uawebservicedemo.models.company.skillclassification;

import org.eclipse.milo.opcua.stack.core.StatusCodes;
import org.eclipse.milo.opcua.stack.core.UaRuntimeException;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.opcfoundation.uawebservicedemo.database.EmployeeDatabase;
import org.opcfoundation.uawebservicedemo.database.entity.Skill;
import org.opcfoundation.uawebservicedemo.database.mapper.SkillMapper;
import org.opcfoundation.uawebservicedemo.models.EmployeeTwinSpace;
import org.opcfoundation.webserver.digitaltwin.submodel.DynamicSubmodelType;
import org.opcfoundation.webserver.service.message.digitaltwin.GetObjectElementListRequest;
import org.opcfoundation.webserver.service.message.digitaltwin.GetObjectElementListResponse;
import org.opcfoundation.webserver.types.digitaltwin.ObjectElementDescriptor;

import java.util.List;
import java.util.concurrent.CompletableFuture;

public class SkillClassificationSubmodelType extends DynamicSubmodelType {
    public SkillClassificationSubmodelType(EmployeeTwinSpace twinSpace) {
        super("SkillClassificationSubmodelType", new LocalizedText("SkillClassificationModel"), twinSpace);
    }

    // Get skill information of employee
    @Override
    public CompletableFuture<GetObjectElementListResponse> onGetObjectElementList(GetObjectElementListRequest request)
    {
        try
        {
            // Database query
            SkillMapper skillMapper = EmployeeDatabase.getSkillMapper();
            List<Skill> skills = skillMapper.getAllSkill();

            // Return general information of skill
            SkillClassType skillClassType = ((EmployeeTwinSpace)nodeManager).skillClassType;
            GetObjectElementListResponse response = new GetObjectElementListResponse();

            for (Skill item : skills)
            {
                ObjectElementDescriptor descriptor = new ObjectElementDescriptor(
                        item.getId().toString(), // SkillId
                        new LocalizedText(item.getName()), // SkillName
                        skillClassType); // SkillType

                response.add(descriptor);
            }

            return CompletableFuture.completedFuture(response);
        } catch (Exception e) {
            throw new UaRuntimeException(StatusCodes.Bad_ResourceUnavailable);
        }
    }
}

File: opcua-webservice-node/examples/demoserver/src/models/company/skill/SkillClassificationSubmodelType.ts
import { UaLocalizedText } from "opcua-webapi-ts";
import { SubmodelTypeDynamic, GetObjectElementListRequest, GetObjectElementListResponse, ObjectElementDescriptor } from "opcua-webservice-node";

import { prisma } from "../../../connectors/prismaClient";
import { EmployeeTwinSpace } from "../../EmployeeTwinSpace";
import type { SkillClassType } from "./SkillClassType";

export class SkillClassificationSubmodelType extends SubmodelTypeDynamic {
    constructor(skillClassType: SkillClassType, twinSpace: EmployeeTwinSpace) {
        super("SkillClassificationSubmodelType", new UaLocalizedText("SkillClassificationModel"), twinSpace);
        this.mayAdd(skillClassType);
    }
        
    override async onGetObjectElementList(request: GetObjectElementListRequest): Promise<GetObjectElementListResponse> {
        // Database query
        const skills = await prisma.skill.findMany({
            select: {
                ID: true,
                SkillName: true
            }
        });

        // Return general information of skill        
        let response = new GetObjectElementListResponse();

        for (const item of skills) {
            const descriptor = new ObjectElementDescriptor(
                    item.ID.toString(), // SkillId
                    new UaLocalizedText(item.SkillName), // SkillName
                    EmployeeTwinSpace.SkillClassType); // SkillType

            response.add(descriptor);
        }

        return response;       
    }
}